packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -427,6 +430,9 @@ function isOasString(type: Type): boolean {
   } else if (type.kind === "Union") {
     // A union where all variants are an OasString
     return type.options.every((o) => isOasString(o));
+  } else if (type.kind === "UnionVariant") {
+    // A union variant where the type is an OasString
+    return isOasString(type.type);
```

Comment:
as we will use union to express extensible enum, there's a case in cadl ranch to directly use union variant as the discriminator value type.